process_pr.py

```diff
@@ -1242,50 +1320,134 @@ def process_pr(repo_config, gh, repo, issue, dryRun, cmsbuild_user=None, force=F
+                        "files": sorted(
+                            x["filename"] for x in get_commit(repo.full_name, commit.sha)["files"]
+                        ),
+                    }
+
+                cache_entry = commit_cache[commit.sha]
+                events[datetime.fromtimestamp(cache_entry["time"])] = {
+                    "type": "commit",
+                    "value": cache_entry["files"],
+                }
+
+        print("Saving commit cache")
+        old_body = cache_comment.body if cache_comment else CMSBOT_TECHNICAL_MSG
+        new_body = (
+            REGEX_COMMITS_CACHE.sub("", old_body)
+            + "<!-- commits cache: "
+            + dumps_maybe_compress(commit_cache)
+            + " -->"
+        )
+        if len(new_body) <= 65535:
+            if not dryRun:
+                if cache_comment:
+                    if old_body != new_body:
+                        cache_comment.edit(new_body)
+                else:
+                    issue.create_comment(new_body)
+            else:
+                if cache_comment:
+                    print("DRY RUN: Updating existing comment with text")
+                    print(new_body)
+                else:
+                    print("DRY RUN: Creating technical comment with text")
+                    print(new_body)
+        else:
+            raise RuntimeError(f"Updated comment body too long: {len(new_body)} > 65535")
+
+    events = sorted(events.items())
+    import pprint
+
+    print("Events:", pprint.pformat(events))
+    print("Recalculating signatures", signatures)
+
+    # Process commits and signatures
+    for _, event in events:
+        if event["type"] == "sign":
+            selected_cats = event["value"]["selected_cats"]
+            ctype = event["value"]["ctype"]
+            if any(x in signing_categories for x in selected_cats):
+                if ctype == "+1":
+                    for sign in selected_cats:
+                        signatures[sign] = "approved"
+                        if (
+                            (test_comment is None)
+                            and ((repository in auto_test_repo) or ("*" in auto_test_repo))
+                            and sign not in ("code-checks", "tests", "orp")
+                        ):
+                            test_comment = event["value"]["comment"]
+                elif ctype == "-1":
+                    for sign in selected_cats:
+                        signatures[sign] = "rejected"
+            else:
+                print(f"Ignoring event: {selected_cats} includes none of {signing_categories}")
+        elif event["type"] == "commit":
+            if cmssw_repo:
+                # chg_categories = {"orp", "code-checks"}
+                # for fn in event["value"]:
+                #     chg_categories.update(
+                #         get_package_categories(cmssw_file2Package(repo_config, fn))
+                #     )
```

Comment:
@iarspider , shouldn't we reset the signature for the cats touched by this commit?